src/app/grouped-account-select/grouped-account-select.component.ts

```diff
@@ -0,0 +1,311 @@
+              showRecoverSubAccountInput: false,
+              accounts: accounts
+                .map((account, j) => ({
+                  ...account,
+                  ...users[account.publicKey],
+                }))
+                .sort(sortAccounts),
+            });
+
+            // get the first account in the list and set it as the lastUsed.
+            const firstKey = this.accountGroups.keys().next().value;
+            const firstGroup = this.accountGroups.get(firstKey);
+            if (firstGroup) {
+              firstGroup.accounts[0].lastUsed = true;
+              this.accountGroups.set(firstKey, firstGroup);
+            }
+          }
+        );
+      });
+  }
+
+  /**
+   * We need this to address angular's weird default sorting of Maps by key when
+   * iterating in the template. See this issue for details. We just want to
+   * preserve the natural order of the Map entries:
+   * https://github.com/angular/angular/issues/31420
+   */
+  keyValueSort() {
+    return 1;
+  }
+
+  getLoginMethodIcon(loginMethod: LoginMethod = LoginMethod.DESO): string {
+    return {
+      [LoginMethod.DESO]: 'assets/logo-deso-mark.svg',
+      [LoginMethod.GOOGLE]: 'assets/google_logo.svg',
+      [LoginMethod.METAMASK]: 'assets/metamask.png',
+    }[loginMethod];
+  }
+
+  selectAccount(publicKey: string) {
+    this.accountService.updateAccountInfo(publicKey, {
+      lastLoginTimestamp: Date.now(),
+    });
+    this.onAccountSelect.emit(publicKey);
+  }
+
+  hideAccount(groupKey: string, account: AccountViewModel) {
+    // NOTE: if there is at least 1 sub account left in the group after hiding this account,
+    // the user only needs the account number to recover it. If there are no sub accounts left,
+    // the user needs the seed phrase + the account number to recover it.
+    const group = this.accountGroups.get(groupKey) ?? {
+      accounts: [],
+    };
+    // get a copy of the underlying array so we can preview what it looks like when hiding this account
+    const hiddenPreview = group.accounts
+      .slice()
+      .filter((a) => a.accountNumber !== account.accountNumber);
+
+    const dialogRef = this.dialog.open(RemoveAccountDialogComponent, {
+      data: {
+        publicKey: account.publicKey,
+        accountNumber: account.accountNumber,
+        username: account.username,
+        isLastAccountInGroup: hiddenPreview.length === 0,
+      },
+    });
+
+    dialogRef.afterClosed().subscribe((confirmed) => {
+      if (confirmed) {
+        this.accountService.updateAccountInfo(account.publicKey, {
+          isHidden: true,
+        });
+        group.accounts = hiddenPreview;
+        this.accountGroups.set(groupKey, group);
+      }
+    });
+  }
+
+  addSubAccount(
+    rootPublicKey: string,
+    { accountNumber }: { accountNumber?: number } = {}
+  ) {
+    const addedAccountNumber = this.accountService.addSubAccount(
+      rootPublicKey,
+      { accountNumber }
```

Comment:
just a note for myself, I we should change this to make accountNumber a required argument here.